src/Controllers/UserController.php

```diff
@@ -46,57 +64,71 @@ public function goToAddUser(RequestInterface $request)
 
     public function addUser(RequestInterface $request)
     {
-        if ($this->session->get('role') === 'admin') {
-            if ($this->userService->createEntity($request, $this->session)) {
-                $response = new Response(Stream::createFromString(' '), []);
-                $response = $response->withStatus(301);
-                $response = $response->withHeader('Location', 'http://local.quiz.com/user/1');
-
-                return $response;
-            }
-            $this->goToAddUser($request);
+        if ($this->session->get('role') !== 'admin') {
+            return $this->renderer->renderView("login.phtml", $request->getRequestParameters());
         }
 
-        return $this->renderer->renderView("login.phtml", $request->getRequestParameters());
+        if ($this->userService->createEntity($request, $this->session)) {
+            $response = new Response(Stream::createFromString(' '), []);
+            $response = $response->withStatus(301);
+            $response = $response->withHeader('Location', 'http://local.quiz.com/user');
+
+            return $response;
+        }
+
+        $this->goToAddUser($request);
     }
 
     public function deleteUser(RequestInterface $request)
     {
-        if ($this->session->get('role') === 'admin') {
-            if ($this->userService->delete($request)) {
-                $response = new Response(Stream::createFromString(' '), []);
-                $response = $response->withStatus(301);
-                $response = $response->withHeader('Location', 'http://local.quiz.com/user/1');
+        if ($this->session->get('role') !== 'admin') {
+            return $this->renderer->renderView("login.phtml", $request->getRequestParameters());
+        }
 
-                return $response;
-            }
+        if ($this->userService->delete($request)) {
+            $response = new Response(Stream::createFromString(' '), []);
+            $response = $response->withStatus(301);
+            $response = $response->withHeader('Location', 'http://local.quiz.com/user');
 
-            return $this->listAll($request);
+            return $response;
         }
-        return $this->renderer->renderView("login.phtml", $request->getRequestParameters());
+
+        return $this->listAll($request);
```

Comment:
This seems weird.

On a delete request, you are rendering a list of stuff.

what if the HTTP verb is `POST`? you are still rendering the list.

I suggest you redirect the user to the appropriate url instead of calling the controller method.
(same for when user is not admin, you should redirect to login instead of rendering login)